lib/diagnostic.js

```diff
@@ -38,16 +71,36 @@ let owner = null;
 // Methods: vocalize
 
 // Replace null with your answer
-let pet = null;
+let pet = {
+  name: 'Slim Dog',
+  species: 'Dog',
+  breed: 'Labrador',
+  vocalization: 'yes',
+  vocalize: function () {
+    let loud = this.vocalization;
+    if (loud === 'yes') {
+      return 'Woof woof';
+    } else {
+      return 'Grrr';
+    }
+  },
+};
 
 // 4. Create a constructor function for pet objects, as described above. The
 // constructor function should be responsible for setting the values of `name`,
 // `species`, `breed`, and `vocalization`.
 
-const Pet = function () {
-  /* Your answer here */
+const Pet = function (species, breed, vocalization) {
+  this.species = species;
+  this.breed = breed;
+  this.vocalization = vocalization;
 };
 
+const vocalization = function () {
+
+}
```

Comment:
We want ```vocalize()``` to return the value of the object's ```vocalization``` attribute and to be available to all objects of the type ```Pet```.
```javascript
Pet.prototype.vocalize = function() {
  return this.vocalization;
};
```